pkg/handlers/lifecycle/cni/multus/handler.go

```diff
@@ -0,0 +1,190 @@
+func (m *MultusConfig) AddFlags(prefix string, flags *pflag.FlagSet) {
+	m.helmAddonConfig.AddFlags(prefix+".helm-addon", flags)
+}
+
+type MultusHandler struct {
+	client              ctrlclient.Client
+	config              *MultusConfig
+	helmChartInfoGetter *config.HelmChartGetter
+}
+
+var (
+	_ commonhandlers.Named                   = &MultusHandler{}
+	_ lifecycle.AfterControlPlaneInitialized = &MultusHandler{}
+	_ lifecycle.BeforeClusterUpgrade         = &MultusHandler{}
+)
+
+func New(
+	c ctrlclient.Client,
+	cfg *MultusConfig,
+	helmChartInfoGetter *config.HelmChartGetter,
+) *MultusHandler {
+	return &MultusHandler{
+		client:              c,
+		config:              cfg,
+		helmChartInfoGetter: helmChartInfoGetter,
+	}
+}
+
+func (m *MultusHandler) Name() string {
+	return "MultusHandler"
+}
+
+func (m *MultusHandler) AfterControlPlaneInitialized(
+	ctx context.Context,
+	req *runtimehooksv1.AfterControlPlaneInitializedRequest,
+	resp *runtimehooksv1.AfterControlPlaneInitializedResponse,
+) {
+	commonResponse := &runtimehooksv1.CommonResponse{}
+	m.apply(ctx, &req.Cluster, commonResponse)
+	resp.Status = commonResponse.GetStatus()
+	resp.Message = commonResponse.GetMessage()
+}
+
+func (m *MultusHandler) BeforeClusterUpgrade(
+	ctx context.Context,
+	req *runtimehooksv1.BeforeClusterUpgradeRequest,
+	resp *runtimehooksv1.BeforeClusterUpgradeResponse,
+) {
+	commonResponse := &runtimehooksv1.CommonResponse{}
+	m.apply(ctx, &req.Cluster, commonResponse)
+	resp.Status = commonResponse.GetStatus()
+	resp.Message = commonResponse.GetMessage()
+}
+
+func (m *MultusHandler) apply(
+	ctx context.Context,
+	cluster *clusterv1.Cluster,
+	resp *runtimehooksv1.CommonResponse,
+) {
+	clusterKey := ctrlclient.ObjectKeyFromObject(cluster)
+
+	log := ctrl.LoggerFrom(ctx).WithValues(
+		"cluster",
+		clusterKey,
+	)
+
+	// Check if Multus is supported for this cloud provider
+	provider := capiutils.GetProvider(cluster)
+	if provider != "eks" && provider != "nutanix" {
+		log.V(5).Info(
+			"Multus is not supported for this cloud provider. Skipping Multus deployment.",
+		)
+		return
+	}
+
+	log.Info(fmt.Sprintf("Cluster is %s. Checking CNI configuration for Multus deployment.", provider))
+
+	// Read CNI configuration to detect which CNI is deployed
+	varMap := variables.ClusterVariablesToVariablesMap(cluster.Spec.Topology.Variables)
+
+	cniVar, err := variables.Get[v1alpha1.CNI](
+		varMap,
+		v1alpha1.ClusterConfigVariableName,
+		[]string{"addons", v1alpha1.CNIVariableName}...)
+	if err != nil {
+		if variables.IsNotFoundError(err) {
+			log.V(5).Info("No CNI specified in cluster config. Skipping Multus deployment.")
+			return
+		}
+		log.Error(err, "failed to read CNI configuration from cluster definition")
+		resp.SetStatus(runtimehooksv1.ResponseStatusFailure)
+		resp.SetMessage(fmt.Sprintf("failed to read CNI configuration: %v", err))
+		return
+	}
+
+	// Get socket path for the CNI provider
+	socketPath, err := cni.SocketPath(cniVar.Provider)
```

Comment:
nit: a little clearer what variable is for
```suggestion
	readinessSocketPath, err := cni.SocketPath(cniVar.Provider)
```